MekHQ/src/mekhq/gui/dialog/ForceTemplateAssignmentDialog.java

```diff
@@ -0,0 +1,291 @@
+        Unit unit = unitList.getSelectedValue();
+        
+        currentScenario.removeUnit(unit.getId());
+        currentScenario.addUnit(unit.getId(), templateList.getSelectedValue().getForceName());
+        unit.setScenarioId(currentScenario.getId());
+        MekHQ.triggerEvent(new DeploymentChangedEvent(unit, currentScenario));
+        
+        refreshUnitList();
+    }
+    
+    /**
+     * Event handler for assigning a force to a scenario and specific template
+     */
+    private void assignForceToTemplate() {
+        Force force = forceList.getSelectedValue();
+        int forceID = forceList.getSelectedValue().getId();
+        
+        // all this stuff apparently needs to happen when assigning a force to a scenario
+        campaignGUI.undeployForce(force);
+        force.clearScenarioIds(campaignGUI.getCampaign(), true);
+        force.setScenarioId(currentScenario.getId());
+        currentScenario.addForce(forceID, templateList.getSelectedValue().getForceName());
+        for (UUID uid : force.getAllUnits()) {
+            Unit u = campaignGUI.getCampaign().getUnit(uid);
+            if (null != u) {
+                u.setScenarioId(currentScenario.getId());
+            }
+        }
+        MekHQ.triggerEvent(new DeploymentChangedEvent(force, currentScenario));
+        
+        refreshForceList();
+    }
+        
+    private class UnitListCellRenderer extends DefaultListCellRenderer {
+        @Override
+        public Component getListCellRendererComponent(JList<?> list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
+            Component cmp = super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus);
+            Unit unit = (Unit) value;
+            String cellValue = currentScenario.getPlayerUnitTemplates().containsKey(unit.getId()) ?
+                    String.format("%s (%s)", unit.getName(), currentScenario.getPlayerUnitTemplates().get(unit.getId()).getForceName()) :
+                        unit.getName();
+            ((JLabel) cmp).setText(cellValue);
+            return cmp;
+        }
+    }
+    
+    private class ForceListCellRenderer extends DefaultListCellRenderer {
+        @Override
+        public Component getListCellRendererComponent(JList<?> list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
+            Component cmp = super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus);
+            Force force = (Force) value;
+            String cellValue = currentScenario.getPlayerForceTemplates().containsKey(force.getId()) ?
+                    String.format("%s (%s)", force.getName(), currentScenario.getPlayerForceTemplates().get(force.getId()).getForceName()) :
+                    force.getName();
+            ((JLabel) cmp).setText(cellValue);
+            return cmp;
+        }
+    }
+    
+    private class TemplateListCellRenderer extends DefaultListCellRenderer {
```

Comment:
This inner class does not reference an instance of the enclosing class and can be made static to take less memory and avoid a possible source of memory leaks.